### PR TITLE
Add sprite sheet

### PR DESCRIPTION
**Pull request by Nastormo**

### CODE REVIEW

**Comment by AlexHonor** [comment]
Надо еще GraphicsEngine поменять, чтобы он принимал на отрисовку именно строковые id картинок и шрифтов.

File: Resources.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    "Images": {
+        "CrystalMine": "Resources/Buildings/Mines/CrystalMine/CrystalMine.png"
+    },
+    "SpriteSheet": {
+        "SP-Land": ["/home/lev/city_builder/Resources/Environment/SP-Land.png", 16, 16]
```
Comment (MoonDweller29) [inline Resources.json]: /home/lev :)
That's an absolute path, delete it, please

File: Resources.json [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    },
+    "Fonts": {
+        "Arial_20": ["Arial", 20],
+        "CustomFont_20": ["/home/lev/city_builder/Resources/Font/20636.ttf", 20]
```
Comment (MoonDweller29) [inline Resources.json]: abs path again

File: Sources/ResourceManager.py [hunk 3]
```diff
@@ -35,6 +47,24 @@ def create_img(self, name, path):
...
 
+    def create_sprite_sheet(self, name, path, width, height):
+        if name in self.__spriteSheetDict:
+            raise ValueError(f"SpriteSheet resource {name} exist!")
+
+        image = self.__load_img(path)
+        imgArray = []
+        for y in range(image.get_height() // height):
+            for x in range(image.get_width() // width):
+                print (x, y, x * width, y * height)
```
Comment (MoonDweller29) [inline Sources/ResourceManager.py]: That's a debug print

File: Sources/ResourceManager.py [hunk 4]
```diff
@@ -35,6 +47,24 @@ def create_img(self, name, path):
...
+                imgArray.append(image.subsurface(x * width, y * height, width, height))
+        self.__spriteSheetDict[name] = {
```
Comment (MoonDweller29) [inline Sources/ResourceManager.py]: I guess, It'll be better to create a separate SpriteSheet class, which constructs from image, tile_width, tile_height.
So you'll keep objects of SpriteSheet class in __spriteSheetDict.

File: Resources.json (hunk 1 above)
Comment (Nastormo) [reply Resources.json]: Done

File: Resources.json (hunk 2 above)
Comment (Nastormo) [reply Resources.json]: Done

File: Sources/ResourceManager.py (hunk 3 above)
Comment (Nastormo) [reply Sources/ResourceManager.py]: Done

File: Sources/ResourceManager.py (hunk 4 above)
Comment (Nastormo) [reply Sources/ResourceManager.py]: Done

File: Sources/ResourceManager.py [hunk 5]
```diff
@@ -1,16 +1,41 @@
...
 import pygame
+import json
  
 # @TODO нужно будет сделать систему подгрузки ресурсов
 # 1) Шрифты, картинки, может звуки (потом)
 # 2) Доступно из разных модулей программы (Давайте синглтон)
 # 3) Понять че за дичь с pygame, там ресурсы создают сразу с локальными свойствами (Например, в грифт зашит размер)
 # 4) Подумать о том, что потенциально мы можем грузить данные из zip потом. Просто код под эту замену подготовить.
 
+class SpriteSheet:
```
Comment (MoonDweller29) [inline Sources/ResourceManager.py]: I guess every class should be in a separate file

File: Sources/GraphicsEngine.py [hunk 6]
```diff
@@ -17,8 +18,8 @@ def display_flip(self):
 
     # draw methods
 
-    def draw_image(self, image, position, size):
-        tmp = pygame.transform.scale(image, size)
+    def draw_image(self, name, position, size):
+        tmp = pygame.transform.scale(ResourceManager().get_image(name), size)
```
Comment (MoonDweller29) [inline Sources/GraphicsEngine.py:22]: The same change is supposed to be done with draw_text

File: Sources/ResourceManager.py (hunk 5 above)
Comment (Nastormo) [reply Sources/ResourceManager.py]: Done

File: Sources/GraphicsEngine.py (hunk 6 above)
Comment (Nastormo) [reply Sources/GraphicsEngine.py:22]: Done

**Review by MoonDweller29 — APPROVED** [review]

**Pull request merged**